### PR TITLE
core: add static layout

### PR DESCRIPTION
**Pull request by samouri**
**summary**
adds static-layout to core.

**would love help understanding if this change was ok:**
```diff
- element['style']['display'] = '';
+ setStyle(element, 'display', ''); // eslint-disable-line local/no-style-display
```

I made it because otherwise I was seeing conformance violations:
```
src/core/static-layout.js:140:4: ERROR - [JSC_CONFORMANCE_VIOLATION] Violation: Use getStyle, setStyle, or setStyles in src/core/dom/style.js
  140|     element['style']['display'] = '';
```

**why?**
We'd like to add `applyStaticLayout` to compiler.js, adding it to `#core` helps with us doing that without adding TS typechecking (or ts-ignore) to everything.

### CODE REVIEW

File: src/core/static-layout.js [hunk 1]
```diff
@@ -46,17 +46,21 @@ export function hasNaturalDimensions(tagName) {
...
   if (!naturalDimensions_[tagName]) {
     const doc = element.ownerDocument;
     const naturalTagName = tagName.replace(/^AMP\-/, '');
-    const temp = doc.createElement(naturalTagName);
+    const temp = /** @type {!HTMLElement} */ (
+      doc.createElement(naturalTagName)
+    );
     // For audio, should no-op elsewhere.
-    temp.controls = true;
+    if (naturalTagName === 'audio') {
```
Comment (jridgewell) [inline src/core/static-layout.js:61]: Just use an any cast?

File: src/core/static-layout.js [hunk 2]
```diff
@@ -109,7 +115,7 @@ export function applyStaticLayout(element) {
       toggle(element, false);
       // TODO(jridgewell): Temporary hack while SSR still adds an inline
       // `display: none`
-      element['style']['display'] = '';
+      setStyle(element, 'display', ''); // eslint-disable-line local/no-style-display
```
Comment (jridgewell) [inline src/core/static-layout.js:118]: This is fine, but we could probably just remove it entirely. The "temporary" hack was added 3 years ago, and the SSR has been updated.

File: src/core/static-layout.js [hunk 3]
```diff
@@ -199,19 +205,22 @@ export function applyStaticLayout(element) {
  */
 export function getEffectiveLayout(element) {
   // Return the pre-existing value if layout has already been applied.
-  const completedLayoutAttr = element.getAttribute('i-amphtml-layout');
-  if (completedLayoutAttr) {
-    return parseLayout(completedLayoutAttr);
+  const completedLayout = parseLayout(element.getAttribute('layout'));
```
Comment (jridgewell) [inline src/core/static-layout.js:208]: Why this change? The `i-amphtml-layout`, if set, implies the SSR has happened an no further computation is needed.

File: src/core/static-layout.js (hunk 2 above)
Comment (samouri) [reply src/core/static-layout.js:118]: > could _probably_ just remove it entirely

How sure are you about this? I'm happy to remove it, but don't know anything about the context of why it was added.

File: src/core/static-layout.js (hunk 1 above)
Comment (samouri) [reply src/core/static-layout.js:61]: That'll work too. This felt a bit better to me. Will change to any cast

File: src/core/static-layout.js (hunk 2 above)
Comment (samouri) [reply src/core/static-layout.js]: Done. Approve if you are confident/remember context. Otherwise I'll revert
